### PR TITLE
(SIMP-MAINT) Updated contribution guide

### PR DESCRIPTION
**Pull request by trevor-vaughan**
Added text reflecting the current working state of how changes are
processed through the system.

### CODE REVIEW

File: docs/contributors_guide/Contribution_Procedure.rst [hunk 1]
```diff
@@ -77,6 +80,106 @@ The first commit message should be the following format:
...
+adds a breaking change to the system.
+
+Major contributions must have unit tests that cover all major code paths and
+pay particular attention to edge cases.
+
+Acceptance tests must also be provided that cover the primary usage of the code
+that, at a minimum, test the code in a way that end users would use it.
+
+User facing changes should also contain documentation updates that cover the
+expected use cases.
```
Comment (lnemsick-simp) [inline docs/contributors_guide/Contribution_Procedure.rst]: Is this restricted to documentation updates within the module itself or are you expecting simp-doc documentation updates as well?   If simp-doc updates are required, can they be in a deferred ticket?

Comment (trevor-vaughan) [reply docs/contributors_guide/Contribution_Procedure.rst]: @lnemsick-simp I was thinking either one honestly but `simp-doc` is user facing so that would generally fall under something that needs a review while the code for building `simp-doc` is not.

Most minor changes to `simp-doc` would fall under either Trivial or 'requires two person review' from what I've seen.

File: docs/contributors_guide/Contribution_Procedure.rst [hunk 2]
```diff
@@ -77,15 +80,125 @@ The first commit message should be the following format:
...
+and a link in the PR if the acceptance test addition is deferred.
+
+Minor contributions require **one maintenance team member** review. The
+reviewing team member may decide that acceptance tests are required based on
+the understandability of the contribution.
+
+Major Contributions
+"""""""""""""""""""
+
+Major contributions are any changes that affect multiple parts of the system,
```
Comment (lnemsick-simp) [inline docs/contributors_guide/Contribution_Procedure.rst]: On my previous team, two-person reviews were also required when the risk of the solution being correct was high, because of the inexperience of the contributor  WRT the subject matter.  This provided both a sanity check and a mechanism to bring folks up to speed in areas new to them.

File: docs/contributors_guide/Contribution_Procedure.rst (hunk 1 above)
Comment (lnemsick-simp) [reply docs/contributors_guide/Contribution_Procedure.rst]: The update you made addresses the point I was making.

File: docs/conf.py [hunk 3]
```diff
@@ -220,7 +220,12 @@ def setup(app):
     r'^http[s]?:\/\/pgp\.mit\.edu(/.*|$)',
     # These are for the templates in the release emails
     r'.*VERSION.*Upgrade_SIMP.*',
-    r'.*VERSION-.*NUM.*Changelog\.html'
+    r'.*VERSION-.*NUM.*Changelog\.html',
+    # Ignore github searches since they'll cause the API to timeout
+    r'^http[s]:\/\/github\.com\/search\?.*',
+    # FIXME: Puppet is redirecting permanently instead of temporarily which
+    # they need to fix
+    r'^http[s]:\/\/puppet.com\/docs\/.+\/latest\/.+'
```
[the remark this answers is not in the document]
Comment (trevor-vaughan) [reply docs/conf.py]: @jeefberkey I'd like to. It's too much trouble to keep changing the docs all over the place when something changes upstream. They have a selector for the version that you happen to be using.
